weblate/urls.py

```diff
@@ -64,11 +64,17 @@
 # URL regexp for project
 PROJECT = r'(?P<project>[^/]+)/'
 
+COMPONENT = r'(?P<component>[^/]+)/'
+
 # URL regexp for component
-COMPONENT = PROJECT + r'(?P<component>[^/]+)/'
+COMPONENT_SCOPED_BY_PROJECT = PROJECT + COMPONENT
+
+# Hey Chat: Any idea where to put this? This seems a bit sluggish for  a location
+def optional(route_part):
+  return '(?:' + route_part + r')?'
 
 # URL regexp for translations
-TRANSLATION = COMPONENT + LANGUAGE + '/'
+TRANSLATION = optional(PROJECT) + optional(COMPONENT) + LANGUAGE + '/'
```

Comment:
None of existing URLs will work with removed parts. Also having both project and component optional introduces ambiguities:

- how to distinguish between /project/language/ and /component/language/
- the component slugs are unique in project only, so having /component/language/ can match several components